src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -16,19 +19,38 @@ internal enum WebHeaderCollectionType : byte
         WebResponse,
     }
 
-    public sealed class WebHeaderCollection : IEnumerable
+    [Serializable]
+    public class WebHeaderCollection : NameValueCollection, ISerializable
     {
         private const int ApproxAveHeaderLineSize = 30;
         private const int ApproxHighAvgNumHeaders = 16;
 
-        // Lazily initialized fields.
-        private List<string> _entriesList;
-        private Dictionary<string, string> _entriesDictionary;
-        private string[] _allKeys;
+        // Lazily initialized field.
+        private NameValueCollection _innerCollection;
+        private NameValueCollection InnerCollection
+        {
+            get
+            {
+                if (_innerCollection == null)
+                    _innerCollection = new NameValueCollection(ApproxHighAvgNumHeaders, CaseInsensitiveAscii.StaticInstance);
+                return _innerCollection;
+            }
+        }
 
         // This is the object that created the header collection.
         private WebHeaderCollectionType _type;
 
+        protected WebHeaderCollection(SerializationInfo serializationInfo, StreamingContext streamingContext)
+        {
+            int count = serializationInfo.GetInt32("Count");
```

Comment:
@CIPop Ok , i will add  back logging to all methods in WebHeaderCollection then.  GlobalLog is available , so should be straight  forward.